src/coreComponents/mesh/Perforation.cpp

```diff
@@ -59,8 +73,59 @@ void Perforation::postInputInitialization()
   GEOS_ERROR_IF( m_distanceFromHead <= 0,
                  getWrapperDataContext( viewKeyStruct::distanceFromHeadString() ) <<
                  ": distance from well head to perforation cannot be negative." );
-}
 
+  // Setup perforation status function
+  FunctionManager & functionManager = FunctionManager::getInstance();
+
+  if( m_perfStatusTableName.empty() )
+  {
+    // No table function provide, defaults name to perforation name
+    m_perfStatusTableName=getName();
+  }
+  else
+  {
+    // Table name provide as input , check that it exists
+    GEOS_THROW_IF( !functionManager.hasGroup< TableFunction >( m_perfStatusTableName ),
+                   GEOS_FMT( "Perforation status table function missing  : {}", m_perfStatusTableName ),
+                   InputError );
+  }
+
+  if( !functionManager.hasGroup< TableFunction >( m_perfStatusTableName ) )
+  {
+    //  Create the time-dependent perforation status table
+    TableFunction * tableFunction = dynamicCast< TableFunction * >( functionManager.createChild( "TableFunction", m_perfStatusTableName ) );
+
+    array1d< array1d< real64 > > timeCoord;
+    timeCoord.resize( 1 );
+    array1d< real64 > values;
+
+
+    if( m_perfStatusTable.size( 0 ) == 0 )
+    {
+      // No table supplied set all perfs to open
+      real64 alwaysOpen = 1.0;
+      timeCoord[0].emplace_back( 0 );
+      values.emplace_back( alwaysOpen );
+    }
+    else
+    {
+      // User supplied table in Perforation section
+
+      GEOS_THROW_IF( m_perfStatusTable[0].size() != m_perfStatusTable[1].size(),
+                     GEOS_FMT( "Perforation status table missing time or status : {}", getName() ),
```

Comment:
To add the XML file line in the error:
```suggestion
                     GEOS_FMT( "{}: Perforation status table missing time or status.", getDataContext() ),
```